### ISSUE TITLE
Why does some time it takes too long to start testing when using xUnit in AzureDevOps

### ISSUE THREAD

**Issue by harvinders** [xunit/visualstudio.xunit#278]
In the logs below, I could not understand the reason why there is such a huge gap between starting and discovering

[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.1 (64-bit .NET 5.0.7)
[xUnit.net **00:03:15.52**]   Discovering: Drivers.Pacific.Tests
[xUnit.net 00:03:15.61]   Discovered:  Drivers.Pacific.Tests

```
==============================================================================
Task         : .NET Core
Description  : Build, test, package, or publish a dotnet application, or run a custom dotnet command
Version      : 2.187.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/build/dotnet-core-cli
==============================================================================
C:\WINDOWS\system32\chcp.com 65001
Active code page: 65001
Info: .NET Core SDK/runtime 2.2 and 3.0 are now End of Life(EOL) and have been removed from all hosted agents. If you're using these SDK/runtimes on hosted agents, kindly upgrade to newer versions which are not EOL, or else use UseDotNet task to install the required version.
"C:\Program Files\dotnet\dotnet.exe" test C:\agent\_work\27\s\tests\Drivers.Pacific.Tests\Drivers.Pacific.Tests.csproj --logger trx --results-directory C:\agent\_work\_temp --configuration Release --no-build --verbosity normal --collect "XPlat Code Coverage"
Build started 9/07/2021 12:51:23 AM.
Test run for C:\agent\_work\27\s\tests\Resmed.Drivers.Pacific.Tests\bin\Release\net5.0\Drivers.Pacific.Tests.dll (.NETCoreApp,Version=v5.0)
Microsoft (R) Test Execution Command Line Tool Version 16.10.0
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.1 (64-bit .NET 5.0.7)
[xUnit.net 00:03:15.52]   Discovering: Drivers.Pacific.Tests
[xUnit.net 00:03:15.61]   Discovered:  Drivers.Pacific.Tests
[xUnit.net 00:03:15.61]   Starting:    Drivers.Pacific.Tests
``````

**Comment by bradwilson** [comment]
Out of curiosity: how many .dll files are in your output folder? This like it might be #317.

**Comment by bradwilson** [comment]
I'm fixing #317, so I'm going to close this on the assumption that it's a duplicate. Let me know if you're still experiencing this after upgrading to a runner w/ the fix (the next 2.5 prerelease, out soon, will have the fix).
